server/core/region.go

```diff
@@ -313,6 +329,143 @@ func (r *RegionInfo) GetDiffFollowers(other *RegionInfo) []*metapb.Peer {
 	return res
 }
 
+// GetID returns the ID of the region.
+func (r *RegionInfo) GetID() uint64 {
+	return r.meta.GetId()
+}
+
+// GetMeta returns the meta information of the region.
+func (r *RegionInfo) GetMeta() *metapb.Region {
+	return r.meta
+}
+
+// GetApproximateSize returns the approximate size of the region.
+func (r *RegionInfo) GetApproximateSize() int64 {
+	return r.approximateSize
+}
+
+// GetApproximateKeys returns the approximate keys of the region.
+func (r *RegionInfo) GetApproximateKeys() int64 {
+	return r.approximateKeys
+}
+
+// GetDownPeers returns the down peers of the region.
+func (r *RegionInfo) GetDownPeers() []*pdpb.PeerStats {
+	return r.downPeers
+}
+
+// GetPendingPeers returns the pending peers of the region.
+func (r *RegionInfo) GetPendingPeers() []*metapb.Peer {
+	return r.pendingPeers
+}
+
+// GetBytesRead returns the read bytes of the region.
+func (r *RegionInfo) GetBytesRead() uint64 {
+	return r.readBytes
+}
+
+// GetBytesWritten returns the written bytes of the region.
+func (r *RegionInfo) GetBytesWritten() uint64 {
+	return r.writtenBytes
+}
+
+// GetLeader returns the leader of the region.
+func (r *RegionInfo) GetLeader() *metapb.Peer {
+	return r.leader
+}
+
+// GetStartKey returns the start key of the region.
+func (r *RegionInfo) GetStartKey() []byte {
+	return r.meta.StartKey
+}
+
+// GetEndKey returns the end key of the region.
+func (r *RegionInfo) GetEndKey() []byte {
+	return r.meta.EndKey
+}
+
+// GetPeers returns the peers of the region.
+func (r *RegionInfo) GetPeers() []*metapb.Peer {
+	return r.meta.GetPeers()
+}
+
+// GetRegionEpoch returns the region epoch of the region.
+func (r *RegionInfo) GetRegionEpoch() *metapb.RegionEpoch {
+	return r.meta.RegionEpoch
+}
+
+// SetLeader sets the leader of the region.
+// Only for test and simulator.
+func (r *RegionInfo) SetLeader(leader *metapb.Peer) {
```

Comment:
If you want to make `RegionInfo` never be modified after creation, you have to remove all methods that update its fields.